azure-iot-device/azure/iot/device/common/pipeline/operation_flow.py

```diff
@@ -92,13 +92,21 @@ def pass_op_to_next_stage(stage, op):
 
 
 @pipeline_thread.runs_on_pipeline_thread
-def complete_op(stage, op):
+def complete_op(stage, op, error=None):
```

Comment:
>error=None [](start = 27, length = 10)

Although I was advocating for the simplicity of something like this, it now does create two different ways of doing the same thing (and creates the edge case highlighted below). Granted, that edge case did technically already exist, and at least there's now a way to log it.

If anyone feels strongly against this, I would be open to returning to the old method of manually adding error before completion. The most important thing in this PR is establishing the unique paths of expected vs unexpected exceptions.